Content.Server/Emp/EmpSystem.cs

```diff
@@ -0,0 +1,39 @@
+using Content.Server.Explosion.EntitySystems;
+using Content.Server.Light.Components;
+using Content.Server.Light.EntitySystems;
+using Content.Server.Power.Components;
+using Robust.Shared.Map;
+
+namespace Content.Server.Emp;
+
+public sealed class EmpSystem : EntitySystem
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly PoweredLightSystem _poweredLight = default!;
+
+    public const string EmpPulseEffectPrototype = "EffectEmpPulse";
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<EmpOnTriggerComponent, TriggerEvent>(HandleEmpTrigger);
+    }
+
+    public void EmpPulse(EntityCoordinates coordinates, float range, float energyConsumption)
+    {
+        foreach (var uid in _lookup.GetEntitiesInRange(coordinates.ToMap(EntityManager), range))
+        {
+            if (TryComp<BatteryComponent>(uid, out var battery))
+                battery.UseCharge(energyConsumption);
+            if (TryComp<PoweredLightComponent>(uid, out var light))
+                _poweredLight.TryDestroyBulb(light.Owner, light);
+        }
+        Spawn(EmpPulseEffectPrototype, coordinates);
+    }
+
+    private void HandleEmpTrigger(EntityUid uid, EmpOnTriggerComponent comp, TriggerEvent args)
+    {
+        EmpPulse(Transform(uid).Coordinates, comp.Range, comp.EnergyConsumption);
+        args.Handled = true;
```

Comment:
Does this cause problems using entitycoordinates compare to attaching to grid / map, e.g. if the uid is deleted or in a container or whatever.

Reply:
yes it does, if you put it into backpack effect won't show up, or if you run with it in hand, effect will follow you